accession/accession.py

```diff
@@ -564,18 +571,220 @@ def accession_steps(self):
+        "mad_qc_metric": "make_mad_qc_metric",
+        "reads_by_gene_type_qc": "make_reads_by_gene_type_qc",
+    }
+
+    # These properties get added to the GeneTYpeQuantificationQualityMetric, this list needs to be in sync with the portal schema.
+    # The rest will be available to the users via an attachment.
+    GENE_TYPE_PROPERTIES = [
+        "spikein",
+        "rRNA",
+        "Mt_rRNA",
+        "miRNA",
+        "protein_coding",
+        "processed_transcript",
+        "ribozyme",
+        "sRNA",
+        "scaRNA",
+        "snRNA",
+        "snoRNA",
+        "antisense",
+        "sense_overlapping",
+        "sense_intronic",
+    ]
+
+    @property
+    def assembly(self):
+        filekey = "index"
+        return self.find_portal_property_from_filekey(filekey, ASSEMBLY)
+
+    @property
+    def genome_annotation(self):
+        filekey = "index"
+        return self.find_portal_property_from_filekey(filekey, GENOME_ANNOTATION)
+
+    @staticmethod
+    def get_bytes_from_dict(input_dict: Dict, encoding: str = "utf-8") -> bytes:
+        return json.dumps(input_dict).encode(encoding)
+
+    def make_star_mapping_qc(self, encode_bam_file: Dict[str, Any], gs_file: GSFile):
+        if self.file_has_qc(
+            encode_bam_file, "StarQualityMetric"
+        ):  # actual name of the object
+            return
+        qc_file = self.analysis.get_files(
+            filename=gs_file.task.outputs["log_json"]  # task output name
+        )[0]
+        qc = self.backend.read_json(qc_file)
+        star_qc_metric = qc.get("star_log_qc")  # what the key is in actual qc json file
+        del star_qc_metric["Started job on"]
+        del star_qc_metric["Started mapping on"]
+        del star_qc_metric["Finished on"]
+        for key, value in star_qc_metric.items():
+            star_qc_metric[key] = string_to_number(value)
+        qc_bytes = self.get_bytes_from_dict(qc)
+        attachment = self.make_attachment_object(
+            qc_bytes, "text/plain", qc_file.filename, ".txt"
+        )
+        star_qc_metric["attachment"] = attachment
+        return self.queue_qc(
+            star_qc_metric, encode_bam_file, "star-quality-metric"
+        )  # backend mapping adding hyphens and removing caps
+
+    def format_reads_by_gene_type_qc(
+        self, qc_dict: Dict, properties_to_report: List
+    ) -> Dict:
+        output = {prop: qc_dict[prop] for prop in properties_to_report}
+        return output
+
+    def make_reads_by_gene_type_qc(self, encode_file: Dict[str, Any], gs_file: GSFile):
+        if self.file_has_qc(encode_file, "GeneTypeQuantificationQualityMetric"):
+            return
+        qc_file = self.analysis.search_down(gs_file.task, "rna_qc", "rnaQC")[0]
+        qc = self.backend.read_json(qc_file)
+        try:
+            reads_by_gene_type_qc_metric = qc["gene_type_count"]
+        except KeyError:
+            self.logger.exception("Something is wrong with rna_qc file")
```

Comment:
Should be more specific with this error message:
```suggestion
        try:
            gene_type_count_key = "gene_type_count"
            reads_by_gene_type_qc_metric = qc[gene_type_count_key]
        except KeyError:
            self.logger.exception("Could not find key %s in rna_qc file", gene_type_count_key)
```